From 50b29ce228d0622eae2b9854c5e4b9334c540a7a Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Thu, 1 Sep 2021 22:19:19 +0200
Subject: [PATCH] gcc-plugins: modern gcc plugin infrastructure requres c++11
 std

Bump the used std to c++11 to avoid detection failure. This fixes the
support check on modern gcc plugin infrastructure that requires a
minimum standard of c++11 to compile.

This effectively requires GCC 4.8.1.

Signed-off-by: Levente Polyak <levente@leventepolyak.net>
---
 scripts/gcc-plugin.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/gcc-plugin.sh b/scripts/gcc-plugin.sh
index b79fd0bea838..7abc00f60a8b 100755
--- a/scripts/gcc-plugin.sh
+++ b/scripts/gcc-plugin.sh
@@ -8,7 +8,7 @@ srctree=$(dirname "$0")
 gccplugins_dir=$($* -print-file-name=plugin)
 
 # we need a c++ compiler that supports the designated initializer GNU extension
-$HOSTCC -c -x c++ -std=gnu++98 - -fsyntax-only -I $srctree/gcc-plugins -I $gccplugins_dir/include 2>/dev/null <<EOF
+$HOSTCC -c -x c++ -std=gnu++11 - -fsyntax-only -I $srctree/gcc-plugins -I $gccplugins_dir/include 2>/dev/null <<EOF
 #include "gcc-common.h"
 class test {
 public:
-- 
2.33.0

commit 67a5a68013056cbcf0a647e36cb6f4622fb6a470
Author: Valdis Kletnieks <valdis.kletnieks@vt.edu>
Date:   Sat Dec 26 13:21:58 2020 -0500

    gcc-plugins: fix gcc 11 indigestion with plugins...
    
    Fedora Rawhide has started including gcc 11,and the g++ compiler
    throws a wobbly when it hits scripts/gcc-plugins:
    
      HOSTCXX scripts/gcc-plugins/latent_entropy_plugin.so
    In file included from /usr/include/c++/11/type_traits:35,
                     from /usr/lib/gcc/x86_64-redhat-linux/11/plugin/include/system.h:244,
                     from /usr/lib/gcc/x86_64-redhat-linux/11/plugin/include/gcc-plugin.h:28,
                     from scripts/gcc-plugins/gcc-common.h:7,
                     from scripts/gcc-plugins/latent_entropy_plugin.c:78:
    /usr/include/c++/11/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO
     C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
       32 | #error This file requires compiler and library support \
    
    In fact, it works just fine with c++11, which has been in gcc since 4.8,
    and we now require 4.9 as a minimum.
    
    Signed-off-by: Valdis Kletnieks <valdis.kletnieks@vt.edu>
    Acked-by: Josh Poimboeuf <jpoimboe@redhat.com>
    Signed-off-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/82487.1609006918@turing-police

diff --git a/scripts/gcc-plugins/Makefile b/scripts/gcc-plugins/Makefile
index d66949bfeba4..b5487cce69e8 100644
--- a/scripts/gcc-plugins/Makefile
+++ b/scripts/gcc-plugins/Makefile
@@ -22,9 +22,9 @@ always-y += $(GCC_PLUGIN)
 GCC_PLUGINS_DIR = $(shell $(CC) -print-file-name=plugin)
 
 plugin_cxxflags	= -Wp,-MMD,$(depfile) $(KBUILD_HOSTCXXFLAGS) -fPIC \
-		   -I $(GCC_PLUGINS_DIR)/include -I $(obj) -std=gnu++98 \
+		   -I $(GCC_PLUGINS_DIR)/include -I $(obj) -std=gnu++11 \
 		   -fno-rtti -fno-exceptions -fasynchronous-unwind-tables \
-		   -ggdb -Wno-narrowing -Wno-unused-variable -Wno-c++11-compat \
+		   -ggdb -Wno-narrowing -Wno-unused-variable \
 		   -Wno-format-diag
 
 plugin_ldflags	= -shared
